downstreamadapter/eventcollector/event_collector.go

```diff
@@ -206,7 +207,7 @@ func (c *EventCollector) Close() {
 	log.Info("event collector is closed")
 }
 
-func (c *EventCollector) AddDispatcher(target dispatcher.EventDispatcher, memoryQuota uint64, bdrMode bool) {
+func (c *EventCollector) AddDispatcher(target dispatcher.EventDispatcher, memoryQuota uint64, bdrMode bool, redo bool) {
```

Comment:
I prefer to not pass `memoryQuota, bdrMode, redo` as param, we can all get these from the target dispatcher, which will make the function more short and clear.